pypesto/visualize/parameters.py

```diff
@@ -394,8 +394,24 @@ def handle_inputs(
             for inner_xs_idx in inner_xs
         ]
         # set bounds for inner parameters
-        inner_lb = np.full(len(inner_xs_names), -np.inf)
-        inner_ub = np.full(len(inner_xs_names), np.inf)
+        from ..hierarchical.calculator import HierarchicalAmiciCalculator
+
+        # Check if objective has a calculator attribute
+        if hasattr(result.problem.objective, 'calculator') and isinstance(
+            inner_calculator := result.problem.objective.calculator,
+            HierarchicalAmiciCalculator,
+        ):
+            inner_bounds = np.array(
+                [
+                    inner_calculator.inner_problem.xs[inner_name].get_bounds()
+                    for inner_name in inner_xs_names
+                ]
+            )
+            inner_lb = inner_bounds[:, 0]
+            inner_ub = inner_bounds[:, 1]
```

Comment:
There are `0`s and `1`s in this code, which sometimes means lower or upper bound, and sometimes means offset or scaling parameter -- these could be changed to constants for better readability.

Reply:
bounds are now always obtained with UPPER_BOUND and LOWER_BOUND